src/billing/ConsolidatedSpendReport.tsx

```diff
@@ -0,0 +1,427 @@
+              newStore[spendReportLengthInDays] = { spendReport: updatedWorkspaces, startDate, endDate };
+              return newStore;
+            });
+            setOwnedWorkspaces(updatedWorkspaces);
+          }
+        });
+      }
+    };
+    initialize();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [signal, spendReportLengthInDays, allWorkspaces]);
+
+  return (
+    <>
+      <div
+        style={{
+          // color: colors.dark(),
+          fontSize: 18,
+          fontWeight: 600,
+          display: 'flex',
+          alignItems: 'center',
+          marginLeft: '1rem',
+          marginTop: '1rem',
+        }}
+      >
+        Consolidated Spend Report
+      </div>
+      <div style={{ padding: '1.5rem 0 0', flexGrow: 1, display: 'flex', flexDirection: 'column', marginLeft: '1rem' }}>
+        <div
+          style={{
+            display: 'grid',
+            gridTemplateColumns: 'repeat(3, minmax(max-content, 1fr))',
+            rowGap: '1.66rem',
+            columnGap: '1.25rem',
+          }}
+        >
+          <DateRangeFilter
+            label='Date range'
+            rangeOptions={[7, 30, 90]}
+            defaultValue={spendReportLengthInDays}
+            style={{ gridRowStart: 1, gridColumnStart: 1 }}
+            onChange={(selectedOption) => {
+              if (selectedOption !== spendReportLengthInDays) {
+                setSpendReportLengthInDays(selectedOption);
+              }
+            }}
+          />
+          <SearchFilter
+            placeholder='Search by name, project or bucket'
+            style={{ gridRowStart: 1, gridColumnStart: 2, margin: '1.35rem' }}
+            onChange={setSearchValue}
+          />
+        </div>
+        <div aria-live='polite' aria-atomic>
+          <span aria-hidden>*</span>
+          Total spend includes infrastructure or query costs related to the general operations of Terra.
+        </div>
+        {!_.isEmpty(filteredOwnedWorkspaces) && (
+          <>
+            <div role='table' aria-label='owned workspaces'>
+              <ConsolidatedSpendWorkspaceCardHeaders onSort={setWorkspaceSort} sort={workspaceSort} />
+              <div style={{ position: 'relative' }}>
+                {_.flow(
+                  _.orderBy(
+                    [(workspace) => parseCurrencyIfNeeded(workspaceSort.field, _.get(workspaceSort.field, workspace))],
+                    [workspaceSort.direction]
+                  ),
+                  _.map((workspace: GoogleWorkspaceInfo) => {
+                    return (
+                      <ConsolidatedSpendWorkspaceCard
+                        workspace={workspace}
+                        billingAccountDisplayName={workspace.namespace}
+                        billingProject={{
+                          cloudPlatform: 'GCP',
+                          billingAccount: workspace.billingAccount,
+                          projectName: workspace.googleProject,
```

Comment:
`billingProject.projectName` should be `workspace.namespace`. The Google project is a separate entity.